helm_deploy/cla-public/templates/_helpers.tpl

```diff
@@ -61,3 +61,37 @@ Create the name of the service account to use
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{- define "cla-public.app.vars" -}}
+{{- $environment := .Values.environment -}}
+- name: ALLOWED_HOSTS
+  value: "{{ .Values.host }}"
+- name:  CLA_ENV
+  value: "{{ $environment }}"
+{{ range $name, $data := .Values.envVars }}
+- name: {{ $name }}
+{{- if $data.value }}
+  value: "{{ $data.value }}"
+{{- else if $data.secret }}
+  valueFrom:
+    secretKeyRef:
+      name: {{ $data.secret.name }}
+      key: {{ $data.secret.key }}
+      {{- if eq $environment "development" }}
+      optional: true
+      {{- else }}
+      optional: {{ $data.secret.optional | default false }}
```

Comment:
Since we have removed `useDevValues` from each yaml file, can we make remove `$data.secret.optional` since it is no longer being used and make it just be false ?

Same with line 93

Reply:
`useDevValues` was a way of saying "true if on dev, otherwise false"
But if we made your proposed change, we'd be locked in to having every env var set like that, and not be able to make any optional in non-dev environments

Reply:
I thought with that logic in mind, we would want it to always be false if not on dev? Plus we aren't setting an optional key in any of our secrets which is why I thought to remove it.

Reply:
Lines 80-81 mean that on dev it's always optional
Lines 82-83 are for non-dev, so what we do in lines 80-81 isn't relevant to those

You're right we don't currently have any optional secrets on non-dev. But we _could_ set the optional key on our secrets, and I'd prefer a helper that's ready for that as we have it. Making the change you suggest means we'd have to update the helper if we ever wanted that, which personally I don't think is worth it

Reply:
Going to keep as is!